src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
@shyamnamboodiripad @srivatsn @dpoeschl   things became weird because we added Trigger tag. without it, TriggerDescriptor was a special one. now we found round trip issue for trigger descriptor, and we choose to solve it by making special "Trigger" tag. once we added it TriggerDiagnosticDescriptor became a weird thing.

if we go that route, we should remove TriggerDiagnosticDescriptor and just add "Create" method to DiagnosticDescriptor.

about DiagnosticDescriptor being not sealed. I don't know why it is not sealed since its role is providing metadata but making it extensible, we can't trust whether it will follow our contract or not. if there is one thing we would benefit from it being extensible will be if there is someone who want to calculate those data lazily. (like compiler one?)